VContainer/Assets/VContainer/Runtime/Container.cs

```diff
@@ -117,14 +118,42 @@ object CreateTrackedInstance(IRegistration registration)
         IRegistration FindRegistration(Type type)
         {
             IScopedObjectResolver scope = this;
+            
+            CollectionRegistration collectionRegistration = null;
             while (scope != null)
             {
                 if (scope.TryGetRegistration(type, out var registration))
                 {
-                    return registration;
+                    if (registration is CollectionRegistration collection)
+                    {
+                        var elementType = type.GetGenericArguments()[0];
```

Comment:
This is where the allocation for `Type[]` comes in, and we want to avoid it if possible.